cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Node.kt

```diff
@@ -272,9 +271,10 @@ open class Node : IVisitable<Node>, Persistable, LanguageProvider, ScopeProvider
 
     open fun addPrevDFG(
         prev: Node,
-        properties: MutableMap<Properties, Any?> = EnumMap(Properties::class.java)
+        granularity: GranularityType = GranularityType.FULL,
+        partialTarget: Declaration? = null,
```

Comment:
Similar here. I would also consider renaming `partialTarget` to `target` or something similar. At least to me it is confusing why the DFG that can be `FULL` is related to a declaration that is only a partial target.

Reply:
It is only applicable to partial flows. See https://github.com/Fraunhofer-AISEC/cpg/blob/2083d20af8fc0eaf0a104088765bb35e7000d61b/cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/edge/Dataflow.kt#L56-60.

I could add an additional check if granularity is really set to partial if partialTarget is set.

Reply:
I have tried another approach using sealed classes. Please have a look whether this is more intuitive @konradweiss @KuechA